apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/DLQGrpcRdaSource.java

```diff
@@ -0,0 +1,226 @@
+
+  private final DLQDao dao;
+  private final BiPredicate<Long, TMessage> sequencePredicate;
+
+  /**
+   * The primary constructor for this class. Constructs a GrpcRdaSource and opens a channel to the
+   * gRPC service.
+   *
+   * @param config the configuration values used to establish the channel
+   * @param caller the GrpcStreamCaller used to invoke a particular RPC
+   * @param appMetrics the MetricRegistry used to track metrics
+   * @param claimType the claim type
+   */
+  public DLQGrpcRdaSource(
+      EntityManager entityManager,
+      BiPredicate<Long, TMessage> sequencePredicate,
+      RdaSourceConfig config,
+      GrpcStreamCaller<TMessage> caller,
+      MetricRegistry appMetrics,
+      String claimType) {
+    this(
+        entityManager,
+        sequencePredicate,
+        config.createChannel(),
+        caller,
+        config::createCallOptions,
+        appMetrics,
+        claimType);
+  }
+
+  /**
+   * This constructor accepts a fully constructed channel instead of a configuration object. This is
+   * used internally by the primary constructor but is also used by unit tests to allow a mock
+   * channel to be provided.
+   *
+   * @param channel channel used to make RPC calls
+   * @param caller the GrpcStreamCaller used to invoke a particular RPC
+   * @param appMetrics the MetricRegistry used to track metrics
+   */
+  @VisibleForTesting
+  DLQGrpcRdaSource(
+      EntityManager entityManager,
+      BiPredicate<Long, TMessage> sequencePredicate,
+      ManagedChannel channel,
+      GrpcStreamCaller<TMessage> caller,
+      Supplier<CallOptions> callOptionsFactory,
+      MetricRegistry appMetrics,
+      String claimType) {
+    super(
+        Preconditions.checkNotNull(channel),
+        Preconditions.checkNotNull(caller),
+        Preconditions.checkNotNull(claimType),
+        callOptionsFactory,
+        appMetrics);
+    this.dao = new DLQDao(Preconditions.checkNotNull(entityManager));
+    this.sequencePredicate = sequencePredicate;
+  }
+
+  /**
+   * Checks for {@link MessageError}s in the database, then calls the service through the specific
+   * implementation of GrpcStreamCaller provided to our constructor. Cancels the response stream if
+   * reading from the stream is interrupted.
+   *
+   * @param maxPerBatch maximum number of objects to collect into a batch before calling the sink
+   * @param sink to receive batches of objects
+   * @return the number of objects that were successfully processed
+   * @throws ProcessingException wrapper around any Exception thrown by the service or sink
+   */
+  @Override
+  public int retrieveAndProcessObjects(int maxPerBatch, RdaSink<TMessage, TClaim> sink)
+      throws ProcessingException {
+    int totalProcessed = 0;
+
+    MessageError.ClaimType type =
+        claimType.equalsIgnoreCase("fiss")
+            ? MessageError.ClaimType.FISS
+            : MessageError.ClaimType.MCS;
```

Comment:
Maybe the claimType passed to the constructor should be an enum now instead of a simple string since we have some logic that depends on it?  Or maybe there could be a method in MessageError.ClaimType to get the right value from a string (by searching values() or something)?

Reply:
Passing in the enum would explicably tie all sources to the database sink, which may always be the case, but sort of creates an arguably unwanted coupling of the two.  I can add a `from(String)` to the enum.